jetty-slf4j-impl/src/main/java/org/eclipse/jetty/logging/JettyLoggerFactory.java

```diff
@@ -166,4 +175,166 @@ public boolean setLoggerLevel(String loggerName, String levelName)
         jettyLogger.setLevel(level);
         return true;
     }
+
+    @Override
+    public Object getAttribute(String name) throws AttributeNotFoundException
+    {
+        Objects.requireNonNull(name, "Attribute Name");
+
+        switch (name)
+        {
+            case "LoggerNames":
+                return getLoggerNames();
+            case "LoggerCount":
+                return getLoggerCount();
+            default:
+                throw new AttributeNotFoundException("Cannot find " + name + " attribute in " + this.getClass().getName());
+        }
+    }
+
+    @Override
+    public void setAttribute(Attribute attribute) throws AttributeNotFoundException
+    {
+        Objects.requireNonNull(attribute, "attribute");
+        String name = attribute.getName();
+        // No attributes are writable
+        throw new AttributeNotFoundException("Cannot set attribute " + name + " because it is read-only");
+    }
+
+    @Override
+    public AttributeList getAttributes(String[] attributeNames)
+    {
+        Objects.requireNonNull(attributeNames, "attributeNames[]");
```

Comment:
I'd tolerate `null` and return an empty list.

Reply:
That is invalid per spec.
And no OpenJDK internal implementation supports null attributeNames.

Reply:
> That is invalid per spec.

Where do you see this? It's not in `DynamicMBean` and I can point to implementations that tolerate `null`, for example `MBeanServerDelegateImpl`.

Reply:
`MBeanServerDelegateImpl.getAttributes(String[] attributes)` returns all attributes when null is passed in.

Which is nice and all, but pointless.
As the lower level code in JMX would never even call this method if the `String[] attributes` is null.

``` java
MBeanServer mbeanServer = ManagementFactory.getPlatformMBeanServer();
ObjectName objectName = ObjectName.getInstance("java.lang", "type", "OperatingSystem");

AttributeList attributeList = mbeanServer.getAttributes(objectName, null);
for (Attribute attr : attributeList.asList())
{
    System.out.printf("Attribute[%s] = %s%n", attr.getName(), attr.getValue());
}
```

results in ...

```
Exception in thread "main" javax.management.RuntimeOperationsException: Exception occurred trying to invoke the getter on the MBean
	at java.management/com.sun.jmx.interceptor.DefaultMBeanServerInterceptor.getAttributes(DefaultMBeanServerInterceptor.java:660)
	at java.management/com.sun.jmx.mbeanserver.JmxMBeanServer.getAttributes(JmxMBeanServer.java:705)
	at jmx.JmxAccessDemo.main(JmxAccessDemo.java:16)
Caused by: java.lang.IllegalArgumentException: Attributes cannot be null
	... 8 more
```